### PR TITLE
docs: "Writing a Tendermint Core application in Java (gRPC)" guide

### PR DESCRIPTION
**Pull request by climber73**
Refs #3808
* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: docs/guides/java.md [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+            var server = new GrpcServer(app, 26658);
+            server.start();
+            server.blockUntilShutdown();
+        }
+    }
+}
+```
+
+It is the entry point of the application.
+Here we create special object `Environment` which knows where to store state of the application.
```
Comment (melekes) [inline docs/guides/java.md]:
```suggestion
Here we create a special object `Environment`, which knows where to store the application state.
```

File: docs/guides/java.md [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+Then we create and srart gRPC server to handle Tendermint's requests.
```
Comment (melekes) [inline docs/guides/java.md]:
```suggestion
Then we create and start the gRPC server to handle Tendermint Core requests.
```

File: docs/guides/java.md [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+
+Create file `$KVSTORE_HOME/src/main/java/io/example/GrpcServer.java`:
```
Comment (melekes) [inline docs/guides/java.md]:
```suggestion
Create the `$KVSTORE_HOME/src/main/java/io/example/GrpcServer.java` file with the following content:
```

**Comment by climber73** [comment]
@melekes thank you for thorough review, and sorry for my English

**Pull request merged**